contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -166,7 +166,7 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 	// identical to HTLC resolution in the link.
 	event, err := h.Registry.NotifyExitHopHtlc(
 		h.payHash, h.htlcAmt, h.htlcExpiry, currentHeight,
-		hodlChan,
+		hodlChan, nil,
```

Comment:
We still have access to the `OnionBlob` method, so we can re-process it when we need to, and the "supplement" the resolvers like we do for the "true" HTLC amount.

Reply:
I strongly think that should be done in this PR. 

Ideally we would have introduced the complete custom record functionality in one pr. We already deviated from that by only implementing the send part, which I think is not ideal. After this PR gets merged, custom records are still not practically usable because the receive part isn't there.

But let's then at least implement that send part fully and also pass in the custom records from the incoming resolver. This is not optional functionality, but a requirement for consistent operation of the node.

Reply:
There's no code path that will result in us _needing_ that information in this PR.

Reply:
Any records passed in to the invoice registry aren't used in this PR. We also don't expose any of the records over the invoice RPCs yet either.

Reply:
Yes, nothing goes wrong. But my point is that a pr should have a logical scope. This PR implements the send part fully, but the receive part only partially. Sometimes it does and sometimes it doesn't pass the eob to invoice registry, which is inconsistent. I'd rather not merge the custom records receive part at all then.

If you want to keep the scope as it is with a partial implementation of receive, then at least add a clear TODO at this point. This is easily forgotten in the follow up.

Reply:
> Yes, nothing goes wrong.

The salient point, there are no fundamental blockers, and no "cannot live with" objections remaining. We have _explicit_ plans for follow ups within the timeframe of our next release.